src/ftest.jl

```diff
@@ -28,6 +35,34 @@ _diffn(t::NTuple{N, T}) where {N, T} = ntuple(i->t[i]-t[i+1], N-1)
 
 _diff(t::NTuple{N, T}) where {N, T} = ntuple(i->t[i+1]-t[i], N-1)
 
+"""
+    ftest(mod::LinearModel)
+
+Prints the F-test against the null model (`Result ~ 1`).
+
+```jldoctest
+julia> dat = DataFrame(Result=[1.1, 1.2, 1, 2.2, 1.9, 2, 0.9, 1, 1, 2.2, 2, 2],
+                       Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2]);
+
+julia> model = lm(@formula(Result ~ 1 + Treatment), dat);
+
+julia> ftest(model.model)
+F-test against the null model:
+F-statistic: 241.62 on 12 and 1 degrees of freedom, p-value: 0.0502
+```
+"""
+function ftest(mod::LinearModel)
+    rss = deviance(mod)
+    tss = nulldeviance(mod)
+
+    n = nobs(mod)
+    p = dof(mod) - 2 # -2 for dispersion parameter and intercept
```

Comment:
Maybe this method should check that `mod` includes the intercept? Otherwise I'm afraid results will be misleading.